### PR TITLE
[BEAM-641] Travis: verify the generated examples and starter archetypes projects can be built.

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @lukecwik @kennknowles

File: .travis/test_archetype.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# This script generates Beam archetypes examples and starter projects,
+# and verifies they can be built with 'mvn clean install'.
+#
+# This script should be executed after all other travis verifications,
+# because it will generate artifacts and modifies pom files.
+
```
Comment (lukecwik) [inline .travis/test_archetype.sh]: Use the same code in test_wordcount.sh to pull out the version
VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -v '\[')

File: .travis/test_archetype.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+mvn archetype:generate -pl sdks/java \
+  -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
+  -DarchetypeGroupId=org.apache.beam \
+  -DarchetypeVersion=0.3.0-incubating-SNAPSHOT \
+  -DgroupId=com.example \
+  -DartifactId=test-beam-archetypes-examples \
+  -Dversion="0.3.0-incubating-SNAPSHOT" \
```
Comment (lukecwik) [inline .travis/test_archetype.sh]: this should just be an arbitrary version like 0.1

File: .travis/test_archetype.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+mvn clean install -pl sdks/java/test-beam-archetypes-examples
+
+mvn archetype:generate -pl sdks/java \
+  -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-starter \
+  -DarchetypeGroupId=org.apache.beam \
+  -DarchetypeVersion=0.3.0-incubating-SNAPSHOT \
+  -DgroupId=com.starter \
+  -DartifactId=test-beam-archetypes-starter \
+  -Dversion="0.3.0-incubating-SNAPSHOT" \
```
Comment (lukecwik) [inline .travis/test_archetype.sh]: arbitrary version like 0.1

File: .travis/test_archetype.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# and verifies they can be built with 'mvn clean install'.
+#
+# This script should be executed after all other travis verifications,
+# because it will generate artifacts and modifies pom files.
+
+mvn archetype:generate -pl sdks/java \
+  -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
+  -DarchetypeGroupId=org.apache.beam \
+  -DarchetypeVersion=0.3.0-incubating-SNAPSHOT \
+  -DgroupId=com.example \
```
Comment (lukecwik) [inline .travis/test_archetype.sh]: org.apache.example?

File: .travis/test_archetype.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  -DinteractiveMode=false \
+  -Dpackage=org.apache.beam.examples
+
+mvn clean install -pl sdks/java/test-beam-archetypes-examples
+
+mvn archetype:generate -pl sdks/java \
+  -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-starter \
+  -DarchetypeGroupId=org.apache.beam \
+  -DarchetypeVersion=0.3.0-incubating-SNAPSHOT \
+  -DgroupId=com.starter \
```
Comment (lukecwik) [inline .travis/test_archetype.sh]: org.apache.starter?

File: .travis/test_archetype.sh (hunk 1 above)
Comment (peihe) [inline .travis/test_archetype.sh]: done

File: .travis/test_archetype.sh (hunk 2 above)
Comment (peihe) [inline .travis/test_archetype.sh]: done

File: .travis/test_archetype.sh (hunk 5 above)
Comment (peihe) [inline .travis/test_archetype.sh]: done

File: .travis/test_archetype.sh (hunk 3 above)
Comment (peihe) [inline .travis/test_archetype.sh]: done

**Comment by peihe** [comment]
PTAL

**Comment by lukecwik** [comment]
Looks like travis is still unhappy, to worrisome points are:
Archetype repository not defined. Using the one from [org.apache.beam:beam-sdks-java-maven-archetypes-examples:0.2.0-incubating] found in catalog remote

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:2.4:generate (default-cli) on project beam-sdks-java-parent: A Maven 2 project already exists in the directory /Users/travis/build/apache/incubator-beam/sdks/java/test-beam-archetypes-examples -> [Help 1]

**Comment by lukecwik** [comment]
Please close PR and re-open when this is ready.

**Comment by kennknowles** [comment]
R: -kennknowles

Seems like I can bow out of this one. +1 to the goals.

**Pull request closed without merging**